future-apps/payroll/test/contracts/Payroll_rates.test.js

```diff
@@ -0,0 +1,142 @@
+  const ONE = bigExp(1, 18)
+  const SIG = '00'.repeat(65) // sig full of 0s
+
+  const formatRate = (n) => bn(n.toFixed(18)).times(ONE)
+
+  const increaseTime = async seconds => {
+    await payroll.mockIncreaseTime(seconds)
+    await priceFeed.mockIncreaseTime(seconds)
+  }
+
+  before('deploy contracts and tokens', async () => {
+    ({ dao, finance, vault, payrollBase } = await deployContracts(owner))
+
+    USD = '0x0000000000000000000000000000000000000001'
+    ETH = '0x0000000000000000000000000000000000000000'
+    await finance.deposit(ETH, bn(50, 18), 'Initial ETH deposit', { from: owner, value: bn(50, 18) })
+
+    DAI = await deployErc20TokenAndDeposit(owner, finance, 'DAI', TOKEN_DECIMALS)
+    ANT = await deployErc20TokenAndDeposit(owner, finance, 'ANT', TOKEN_DECIMALS)
+  })
+
+  beforeEach('initialize payroll app with USD as denomination token', async () => {
+    ({ payroll, priceFeed } = await createPayrollAndPriceFeed(dao, payrollBase, owner, NOW))
+    await payroll.initialize(finance.address, USD, priceFeed.address, RATE_EXPIRATION_TIME, { from: owner })
+  })
+
+  describe('rates', () => {
+    beforeEach('set rates and allow tokens', async () => {
+      const DAI_rate = bn(1)
+      const ANT_rate = bn(0.5)
+      const ETH_rate = bn(20)
+
+      await priceFeed.update(ETH, USD, formatRate(ETH_rate), NOW, SIG)
+      await priceFeed.update(DAI.address, USD, formatRate(DAI_rate), NOW, SIG)
+      await priceFeed.update(ANT.address, USD, formatRate(ANT_rate), NOW, SIG)
+
+      await payroll.addAllowedToken(ETH, { from: owner })
+      await payroll.addAllowedToken(DAI.address, { from: owner })
+      await payroll.addAllowedToken(ANT.address, { from: owner })
+    })
+
+    beforeEach('add employee with salary 1 USD per second', async () => {
+      const salary = 1
+      const receipt = await payroll.addEmployee(employee, salary, 'Boss', await payroll.getTimestampPublic())
+      employeeId = getEventArgument(receipt, 'AddEmployee', 'employeeId')
+    })
+
+    beforeEach('increase time two minutes', async () => {
+      await increaseTime(TWO_MINUTES)
+    })
+
+    context('when the employee requests only ETH', () => {
+      beforeEach('set token allocation', async () => {
+        await payroll.determineAllocation([ETH], [100], { from: employee })
+      })
+
+      it('receives the expected amount of ETH', async () => {
+        const previousETH = await web3.eth.getBalance(employee)
+        const previousDAI = await DAI.balanceOf(employee)
+        const previousANT = await ANT.balanceOf(employee)
+
+        const { tx, receipt } = await payroll.payday(PAYMENT_TYPES.PAYROLL, 0, { from: employee })
+        const { gasPrice } = await web3.eth.getTransaction(tx)
+        const txCost = gasPrice.mul(receipt.gasUsed)
+
+        const currentETH = await web3.eth.getBalance(employee)
+        assert.equal(currentETH.minus(previousETH).plus(txCost).toString(), 6, 'expected current ETH amount does not match')
```

Comment:
Here and elsewhere, let's comment how we got to the expected payout number (e.g. `120s * 1usd/s / 20eth/usd == 6eth`)